### PR TITLE
grt: complete route flow before failing due to congestion

### PR DESCRIPTION
**Pull request by eder-matheus**
Fixes https://github.com/The-OpenROAD-Project/OpenROAD/issues/6073.

* When the design has congestion, grt will complete its flow and save the guides in odb before the error.
* The -allow_congestion flag is preserved for cases where the user wants to go through drt even with congestion in grt.
* odb::dbGuide was updated with the is_congested flag. This is used inside drt to ensure that it will not route a design that has global routing congestion. However, when the -allow_congestion flag is used in grt, this flag is set to false.

### CODE REVIEW

**Comment by eder-matheus** [comment]
@maliberty This is ready for review.

File: src/odb/src/db/dbJournal.cpp [hunk 1]
```diff
@@ -1544,18 +1544,21 @@ void dbJournal::undo_deleteObject()
...
       uint via_layer_id;
+      bool is_congested;
       _log.pop(net_id);
       _log.pop(x_min);
       _log.pop(y_min);
       _log.pop(x_max);
       _log.pop(y_max);
       _log.pop(layer_id);
       _log.pop(via_layer_id);
+      _log.pop(is_congested);
```
Comment (maliberty) [inline src/odb/src/db/dbJournal.cpp:1555]: I don't see this being pushed in the destroy method

File: src/grt/src/GlobalRouter.cpp [hunk 2]
```diff
@@ -4771,7 +4759,7 @@ std::vector<Net*> GlobalRouter::updateDirtyRoutes(bool save_guides)
         saveCongestion();
         logger_->error(GRT,
                        232,
-                       "Routing congestion too high. Check the congestion "
+                       "Routing congestion too high. Check the congested "
```
Comment (maliberty) [inline src/grt/src/GlobalRouter.cpp:4762]: I think `congestion` is better

**Review by maliberty — CHANGES_REQUESTED** [review]

File: src/odb/src/db/dbJournal.cpp (hunk 1 above)
Comment (eder-matheus) [reply src/odb/src/db/dbJournal.cpp:1555]: Where exactly it should be pushed? I was following the same logic I did with via_layer_id and I'm not sure where whould I do that.

Comment (maliberty) [reply src/odb/src/db/dbJournal.cpp:1555]: dbGuide::destroy

Comment (maliberty) [reply src/odb/src/db/dbJournal.cpp:1555]: You did add 
```
    block->_journal->pushParam(_guide->via_layer_);
```
previously.

Comment (eder-matheus) [reply src/odb/src/db/dbJournal.cpp:1555]: Thanks! Fixed it now.

**Comment by maliberty** [comment]
Fix "Check That ODB Files Are Generated "

**Comment by eder-matheus** [comment]
> Fix "Check That ODB Files Are Generated "

Fixed.

**Review by maliberty — APPROVED** [review]

**Pull request merged**